src/dbl_linked_list.py

```diff
@@ -0,0 +1,125 @@
+"""Create a Double Linked List Data Structure."""
+
+
+class Dbl_Node(object):
+    """Create Node objects for use in a Linked List data structure.
+
+    Attributes:
+    value:      A value stored in the Node.
+    nxt:        Pointer to the next Node, used in a Linked List or Stack
+                data structure.
+    """
+    def __init__(self, value=None, nxt=None, prev=None):
+        """Initialize a Node type object."""
+        self.value = value
+        self.nxt = nxt
+        self.prev = prev
+
+
+class Dbl_Linked_List(object):
+    """Double Linked List style Data Structure
+
+        Attributes:
+        head:       Always the most recent item added to a Linked List.
+        _size:      The length of the Linked List or number of Nodes stored
+                    in the list.
+
+        Methods:
+
+        push(value):    Given a value, add a new Node object to a Linked List.
+
+        append(value):  Given a value, adds it at the end of the line.
+
+        pop():          Pop the head Node off the list, return the value.
+
+        shift():        will remove the last value from the line and return it.
+
+        remove(val):    Given a Node, remove that Node from the Linked List.  Else,
+                        raise a ValueError.
+        """
+
+    def __init__(self, maybe_an_iterable=None):
+        """Intialize a doubly linked List Object."""
+        self.head = None
+        self.tail = None
+        self._size = 0
+        if maybe_an_iterable:
+            try:
+                for value in maybe_an_iterable:
+                    self.push(value)
+                nodes = [node for node in self._iterate_from(self.head)]
+                self.tail = nodes[-1]
+            except TypeError:
+                self.push(maybe_an_iterable)
+
+    def push(self, value):
+        """Add a node at the beginning of the line."""
+        self.head = Dbl_Node(value, self.head)
+        self._size += 1
+        if self._size == 1:
+            self.tail = self.head
+        elif self._size == 2:
+            nodes = [node for node in self._iterate_from(self.head)]
+            self.tail = nodes[-1]
+            nodes[-1].prev = self.head
+        else:
+            self.head.nxt.prev = self.head
+
+    def pop(self):
+        """Pop the first value off the head of LL and return it."""
+        if self.head is None:
+            raise IndexError("Cannot pop from an empty Linked List.")
+        val = self.head.value
+        self.head = self.head.nxt
+        self.head.prev = None
```

Comment:
Here's a big vulnerability in your program. Your tests don't cover the scenario where you pop from a list when there is only one item in it.

Consider also that you need to test and account for a similar situation with `shift`!